Source/HTTP/httpcall.cpp

```diff
@@ -449,10 +451,21 @@ try
     retryContext->outerAsyncBlock = asyncBlock;
     retryContext->outerQueue = asyncBlock->queue;
     retry_context* rawRetryContext = static_cast<retry_context*>(shared_ptr_cache::store<retry_context>(retryContext));
+    if (rawRetryContext == nullptr)
+    {
+        XAsyncComplete(asyncBlock, E_HC_NOT_INITIALISED, 0);
+        return E_HC_NOT_INITIALISED;
+    }
```

Comment:
if you don't check that the singleton is alive before duplicating the handle, you are leaking a handle here.  It's supposed to be cleaned up after the async work is completed, but you are not kicking it off even in this case. This check should be at the beginning of HCHttpCallPerformAsync